src/cli/generator/generator.rs

```diff
@@ -95,18 +95,31 @@ impl Generator {
     async fn resolve_io(&self, config: Config<Resolved>) -> anyhow::Result<Vec<Input>> {
         let mut input_samples = vec![];
 
-        let reader = ResourceReader::cached(self.runtime.clone());
-        let proto_reader = ProtoReader::init(reader.clone(), self.runtime.clone());
+        let cached_reader = ResourceReader::cached(self.runtime.clone());
+        let direct = Direct::init(self.runtime.clone());
+        let proto_reader = ProtoReader::init(cached_reader.clone(), self.runtime.clone());
         let output_dir = Path::new(&config.output.path.0)
             .parent()
             .unwrap_or(Path::new(""));
 
         for input in config.inputs {
             match input.source {
-                Source::Curl { src, field_name, headers: resolved_headers } => {
+                Source::Curl { src, field_name, headers, body } => {
                     let url = src.0;
-                    let mut request = reqwest::Request::new(reqwest::Method::GET, url.parse()?);
-                    if let Some(headers_inner) = resolved_headers.headers() {
+                    let body = body.unwrap_or_default();
+
+                    let (request_method, operation_type) = match body {
```

Comment:
User should be able to control if the request is a mutation vs a query. Let's also give them the ability to specify the HTTP method.